tutorials/counter/README.md

```diff
@@ -0,0 +1,252 @@
+<div id="top"></div>
+
+<div align="center">
+  <a href="https://github.com/FuelLabs/sway-applications">
+    <img src="../../logo.png" alt="Logo" width="700" height="170">
+  </a>
+
+  <h3 align="center">React Counter</h3>
+
+  <p align="center">
+    Sway Counter Smart Contract with React and Fuel v2 TypeScript SDK
+    <br />
+    <a  href="https://fuellabs.github.io/fuels-ts/"><strong>Explore the Fuel TypeScript SDK docs »</strong></a>
+    <br />
+    Other Example Apps:
+    <br />
+    <br />
+    <a href="https://github.com/FuelLabs/swayswap"><strong>⚡️ SwaySwap ⚡️</strong></a>
+    <br />
+    <a href="https://github.com/FuelLabs/block-explorer-v2"><strong>Block explorer frontend for Fuel</strong></a>
+  </p>
+</div>
+
+## Table of contents
+
+- [SDK documentation](https://fuellabs.github.io/fuels-ts/)
+- [About](#about)
+  - [Built With](#built-with)
+- [Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Installation](#installation)
+  - [Walkthrough](#walkthrough)
+- [Usage](#usage)
+- [Contributing](#contributing)
+- [License](#license)
+
+## About
+
+![Demo Screen Shot](img/preview.png)
+
+<p align="right">(<a href="#top">back to top</a>)</p>
+
+### Built With
+
+- [Sway](https://fuellabs.github.io/sway/latest/)
+- [fuels-ts](https://fuellabs.github.io/fuels-ts/)
+- [React.js](https://reactjs.org/)
+- [TypeChain](https://github.com/dethcrypto/TypeChain)
+- [TypeScript](https://www.typescriptlang.org/)
+
+<p align="right">(<a href="#top">back to top</a>)</p>
+
+## Getting Started
+
+We recommend that you first read through and understand the basics of [Sway](https://fuellabs.github.io/sway/latest/), the DSL for the Fuel Virtual Machine.
+
+### Prerequisites
+
+Your machine will need to have a few things installed in order to run a local fuel node and this React application:
+
+- [The latest LTS version of Node.js](https://nodejs.org/)
+- [The latest stable Rust toolchain](https://fuellabs.github.io/sway/latest/introduction/installation.html#dependencies)
+- [forc and fuel-core binaries](https://fuellabs.github.io/sway/latest/introduction/installation.html)
+
+### Installation
+
+0. Before beginning, verify that you have completed all [Prerequisite](#prerequisites) installations.
+
+1. Clone the repo
+
+```sh
+git clone https://github.com/FuelLabs/sway-applications.git
+```
+
+2. Open this demo application
+
+```sh
+cd tutorials/counter
+```
+
+3. Install NPM packages
+
+```sh
+npm i
+```
+
+### Walkthrough
+
+This application tutorial consists of five main milestones on your journey to writing your first Sway + React app:
+
+1. Creating a deployable Sway program - in this example we will create a [Smart Contract](https://fuellabs.github.io/sway/latest/sway-program-types/smart_contracts.html)
```

Comment:
Well, yes. A predicate, script, library and contract are all programs since that is a generic term. 
In my opinion I would like to refer to the smart contract explicitly as a program since a library is not _really_ a program since it cannot be run and must be used by something else, I believe a predicate is the same but I'm not familiar enough with them and so that leaves the script however I see the script more as a helper that is used to call a contract or use a library.
That's not a good argument from me nevertheless I think it is cleaner to state that you're creating a counter program. That's just my personal preference.